http/unstable_file_server_test.ts

```diff
@@ -0,0 +1,1069 @@
+  );
+});
+
+Deno.test("serveDir() serves HTTP 304 for if-none-match requests with W/-prefixed etag", async () => {
+  const testurl = "http://localhost/desktop.ini";
+  const fileurl = new URL("./testdata/desktop.ini", import.meta.url);
+  const req1 = new Request(testurl, {
+    headers: { "accept-encoding": "gzip, deflate, br" },
+  });
+  const res1 = await serveDir(req1, serveDirOptions);
+  const etag = res1.headers.get("etag");
+
+  assertEquals(res1.status, 200);
+  assertEquals(res1.statusText, "OK");
+  assertEquals(await Deno.readTextFile(fileurl), await res1.text());
+  assert(typeof etag === "string");
+  assert(etag.length > 0);
+  assert(etag.startsWith("W/"));
+
+  const req2 = new Request(testurl, {
+    headers: { "if-none-match": etag },
+  });
+  const res2 = await serveDir(req2, serveDirOptions);
+
+  assertEquals(res2.status, 304);
+  assertEquals(res2.statusText, "Not Modified");
+  assertEquals("", await res2.text());
+  assert(
+    etag === res2.headers.get("etag") ||
+      etag === "W/" + res2.headers.get("etag"),
+  );
+});
+
+Deno.test("serveDir() resolves path correctly on Windows", {
+  ignore: Deno.build.os !== "windows",
+}, async () => {
+  const req = new Request("http://localhost/");
+  const res = await serveDir(req, { ...serveDirOptions, fsRoot: "c:/" });
+  await res.body?.cancel();
+
+  assertEquals(res.status, 200);
+});
+
+Deno.test(
+  "serveDir() resolves empty sub-directory without asking for current directory read permissions on Windows",
+  {
+    ignore: Deno.build.os !== "windows",
+    permissions: {
+      read: [`${moduleDir}/testdata`],
+      write: true,
+    },
+  },
+  async () => {
+    const tempDir = await Deno.makeTempDir({
+      dir: `${moduleDir}/testdata`,
+    });
+    const req = new Request(`http://localhost/${basename(tempDir)}/`);
+    const res = await serveDir(req, serveDirOptions);
+    await res.body?.cancel();
+
+    assertEquals(res.status, 200);
+
+    await Deno.remove(tempDir);
+  },
+);
+
+Deno.test(async function serveFileHeadRequest() {
+  const req = new Request("http://localhost/testdata/test_file.txt", {
+    method: "HEAD",
+  });
+  const res = await serveFile(req, TEST_FILE_PATH);
+  assert(!res.body);
+  assertEquals(res.status, 200);
+  assertEquals(res.statusText, "OK");
+  assertEquals(res.headers.get("content-type"), "text/plain; charset=UTF-8");
+  assertEquals(res.headers.get("content-length"), "10034");
+});
+
+Deno.test(
+  "`deno serve -R @std/http/unstable-file-server` works",
+  async () => {
+    const port = getAvailablePort();
+    const child = new Deno.Command(Deno.execPath(), {
+      args: [
+        "serve",
+        "-R",
+        `--port=${port}`,
+        "http/unstable_file_server.ts",
+        "--dir-listing",
+        "http/tesdata",
+      ],
+      stdin: "null",
+      stdout: "piped",
+      stderr: "piped",
+    }).spawn();
+    await readUntilMatch(child.stderr, "Listening");
```

Comment:
Surprised to see that the output of `deno serve` prints to `stderr` instead of `stdout`. Is this a bug?